help(bump-cli): sync with the latest changes of README of cli repo (#147)
bump-sh/docs: src/_help/bump-cli.md

- TOC
{:toc}

> The Bump.sh CLI is used to interact with your API documentation or hubs hosted on Bump.sh. With any API definition of your choice (from Swagger, OpenAPI or AsyncAPI), it can help you to:

Author: I'll keep the original form for now if you don't mind (which is also the `help` command text)

- Validate an API document before publishing to your documentation
- Publish an API document to your Bump.sh documentation or hubs
- Compare two API documents to generate a human-readable diff from your API definitions

## Installation

> Bump.sh CLI is a node package currently distributed via NPM which means you need to have Node v14+ interpreter installed on your computer or CI servers.

Reviewer: ```suggestion
The Bump.sh CLI is a node package currently distributed via NPM. This means you must have the Node v14+ interpreter installed on your computer or CI servers.
```

yarn global add bump-cli
```

### Add Bump.sh to your node project

As our CLI is a node package, you can easily embed it to your project by adding the package to your `package.json` file, either with NPM

```shell
npm install --save-dev bump-cli
```

Or with Yarn via

```shell
yarn add --dev bump-cli
```

You can then use any Bump.sh commands with `npx` (same as `npm exec`)

```shell
npx bump --help
```

### How should I do if I'm not using npm ?

Unfortunately, at the moment we only support the Node environment. However, you can download a standalone package directly from the [latest Github release](https://github.com/bump-sh/cli/releases) assets which you can run as a standalone binary. Or you can push your documentation using [our API](https://developers.bump.sh/) (advanced usage only).

## Usage

To list all the available commands, just type `bump` in your command line environment.

```shell
$ bump --help

> The Bump.sh CLI is used to interact with your API documentation hosted on Bump.sh by using the API of developers.bump.sh

Reviewer: ```suggestion
The Bump.sh CLI lets you interact with your API documentation hosted on Bump.sh by using our API (read the docs at [https://developers.bump.sh](https://developers.bump.sh)).
```

Author: This text is the output text of the help command, there's no need to change it here (or we should change the command output too)

VERSION
  bump-cli/2.7.2 linux-x64 node-v16.17.0

USAGE
  $ bump [COMMAND]

COMMANDS
  deploy   Create a new version of your documentation from the given file or URL.
  diff     Get a comparison diff with your documentation from the given file or URL.
  help     Display help for bump.
  preview  Create a documentation preview from the given file or URL.
```

 You can also get some help anytime by adding `--help` to any command. Example: `bump deploy --help`.

## Prepare your Bump.sh account

While some commands don't need any API token (`preview` or `diff`) you will need an access key if you want to interact with your Bump.sh documentation.

> Head to your Documentation settings in the “CI deployment” section, or your Account or Organization settings in the “API keys” section to fetch a personal token for later usage.

Reviewer: ```suggestion
Head over to your Documentation settings in the “CI deployment” section or your Account or Organization settings in the “API keys” section to fetch a personal token for later usage.
```

## Commands

* [`bump deploy [FILE]`](#bump-deploy-file)
* [`bump diff [FILE]`](#bump-diff-file)
* [`bump preview [FILE]`](#bump-preview-file)

### `bump deploy [FILE]`

> When you update your API, you want its documentation to be live for your API users. This is what the deploy command is for.

Reviewer: ```suggestion
When you update your API, you also want its documentation to be up to date for your API users. This is what the deploy command is for.
```

```shell
bump deploy path/to/api-document.yml --doc my-documentation --token $DOC_TOKEN
```

> You can find your own `my-documentation` slug and `$DOC_TOKEN` api key from your [documentation settings](https://bump.sh/docs).
{: .info}

> You can also deploy a given API document to a different branch of your documentation with the `--branch <branch-name>` parameter. Please note the branch will be created if it doesn’t exist. More details about the branching feature is available on [this dedicated help page](https://docs.bump.sh/help/branching). E.g. deploy the API document to the `staging` branch of the documentation:

Reviewer: ```suggestion
You can also deploy a given API document to a different branch of your documentation with the `--branch <branch-name>` parameter. Please note that the branch will be created if it doesn’t exist. More details about the branching feature are available on [this dedicated help page](https://docs.bump.sh/help/branching). E.g., deploy the API document to the `staging` branch of the documentation:
```

Author: ok for the `are` not sure about the other two suggestions

```shell
bump deploy path/to/api-document.yml --doc my-documentation --token $DOC_TOKEN --branch staging
```

#### Deploy a folder all at once

> If you already have a hub in your [Bump.sh](https://bump.sh) account, you can automatically create documentation inside it and deploy to it by publishing a whole directory containing multiple API documents in a single command

Reviewer: ```suggestion
If you already have a hub in your [Bump.sh](https://bump.sh) account, you can automatically create documentation and deploy it into that hub by publishing a whole directory containing multiple API documents in a single command:
```

```shell
bump deploy dir/path/to/apis/ --auto-create --hub my-hub --token $HUB_TOKEN
```

> You can find your own `my-hub` slug and `$HUB_TOKEN` api key from your [hub settings](https://bump.sh/hubs).
{: .info}

> Please note, by default, only files named `{slug}-api.[format]` are published. Where `{slug}` is a name for your API and `[format]` is either `yaml` or `json`. Take into account your own file naming convention by using the `--filename-pattern <pattern>` option.

Reviewer: ```suggestion
Please note that, by default, only files named `{slug}-api.[format]` are published. Where `{slug}` is a name for your API and `[format]` is either `yaml` or `json`. Adjust to your file naming convention using the `--filename-pattern <pattern>` option.
```

Author: I'm not a huge fan of using `that` it's usually an optional word and doesn't help that much in this context imo.

Note that it _can_ include `*` wildcard special character, but **must** include the `{slug}` filter to extract your documentation’s slug from the filename. The pattern can also have any other optional fixed characters.

Here’s a practical example. Let's assume that you have the following files in your `path/to/apis/` directory:

```
path/to/apis
└─ private-api-users-service.json
└─ partner-api-payments-service.yml
└─ public-api-contracts-service.yml
└─ data.json
└─ README.md
```

In order to deploy the 3 services API definition files from this folder (`private-api-users-service.json`, `partner-api-payments-service.yml` and `public-api-contracts-service.yml`), you can execute the following command:

```
bump deploy path/to/apis/ --hub my-hub --filename-pattern '*-api-{slug}-service'
```

#### Validate an API document

> Simulate a deployment of your API document to make sure it is valid by adding the `--dry-run` flag to the `deploy` command, it is particularly useful in a Continuous Integration environment running a test deployment outside your main branch:

Reviewer: ```suggestion
Simulate your API document's deployment to ensure it is valid by adding the `--dry-run` flag to the `deploy` command. It is handy in a Continuous Integration environment running a test deployment outside your main branch:
```

$ bump diff path/to/your/file.yml path/to/your/second_file.yml
* Comparing the two given definition files... done
Modified: GET /consommations
  Response modified: 200
    [Breaking] Body attribute modified: energie
```
> You can create as many diffs as you like without being authenticated. This is a **free and unlimited service** provided as long as you use the service fairly.
{: .info}

_Note: You can also test this feature in our dedicated web application at <https://api-diff.io/>._

#### Authenticated diffs related to your Bump.sh documentation

From an existing Bump.sh documentation, the `diff` command will retrieve a comparison changelog between your latest published documentation and the given file or URL:

```shell
bump diff path/to/your/file.yml --doc my-documentation --token $DOC_TOKEN
```

If you want to compare two unpublished versions of your API document, the `diff` command can retrieve a comparison changelog between two given file or URL, “as simple as `git diff`”:

```shell
bump diff path/to/your/file.yml path/to/your/next-file.yml --doc my-documentation --token $DOC_TOKEN
```

Please check `bump diff --help` for full usage details.

### `bump preview [FILE]`

> When writing a documentation, you might want to preview how it renders on Bump.sh. This is precisely the goal of the `preview` command: it will create a temporary documentation with a unique URL, which will be available for a short period.

Reviewer: ```suggestion
When writing documentation, you might want to preview how it renders on Bump.sh. This is precisely the goal of the `preview` command: it will create temporary documentation with a unique URL, which will be available for a short period.
```

Reviewer: Do we want to be more explicit about the duration?

Usage from a local OpenAPI or AsyncAPI file

```shell
bump preview path/to/file.json
```

> You can also preview a file available from a URL

Reviewer: ```suggestion
You can also preview a file available from a URL:
```
